tests/test_retention_policies.py

```diff
@@ -232,12 +258,27 @@ def test_first_backup(self, server):
 
         # Build a BackupInfo object with status to DONE
         backup_info = build_test_backup_info(
-            server=server, backup_id="test0", end_time=datetime.now(tzlocal())
+            server=server,
+            backup_id="test_backup",
+            end_time=datetime.now(tzlocal()),
+            parent_backup_id=None,
+            children_backup_ids=["test_backup_child"],
         )
 
+        # Build a CHILD BackupInfo object with status to DONE
+        child_backup_info = build_test_backup_info(
+            server=server,
+            backup_id="test_backup_child",
+            end_time=datetime.now(tzlocal()) + timedelta(days=1),
+            parent_backup_id="test_backup",
+            children_backup_ids=None,
+        )
         # instruct the get_available_backups method to return a map with
         # our mock as result and minimum_redundancy = 1
-        server.get_available_backups.return_value = {"test_backup": backup_info}
+        server.get_available_backups.return_value = {
+            "test_backup": backup_info,
+            "test_backup_child": child_backup_info,
+        }
```

Comment:
This doesn't seem to add value to the unit test from what I can see.

I mean, `test_first_backup` simply checks if the `_first_backup` attribute will contain the first `VALID` backup in the catalog. With that in mind, the incremental which you add to the list won't even be taken into account.

Reply:
yup. I agree with this.
the above modifications do not really add something useful to the test (that... is worryingly missing any docstring and difficult to understand) I think we should not apply this code.
@barthisrael @andremagui

Reply:
@gcalacoci @barthisrael  the goal here was just to be more precise in the test. I just think that if you have a get_available_backups.return_value with only ONE backup_info, this will always be the first backup